mailbox/postgres/src/main/java/org/apache/james/mailbox/postgres/mail/MessageRepresentation.java

```diff
@@ -0,0 +1,113 @@
+/****************************************************************
+ * Licensed to the Apache Software Foundation (ASF) under one   *
+ * or more contributor license agreements.  See the NOTICE file *
+ * distributed with this work for additional information        *
+ * regarding copyright ownership.  The ASF licenses this file   *
+ * to you under the Apache License, Version 2.0 (the            *
+ * "License"); you may not use this file except in compliance   *
+ * with the License.  You may obtain a copy of the License at   *
+ *                                                              *
+ *   http://www.apache.org/licenses/LICENSE-2.0                 *
+ *                                                              *
+ * Unless required by applicable law or agreed to in writing,   *
+ * software distributed under the License is distributed on an  *
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY       *
+ * KIND, either express or implied.  See the License for the    *
+ * specific language governing permissions and limitations      *
+ * under the License.                                           *
+ ****************************************************************/
+
+package org.apache.james.mailbox.postgres.mail;
+
+import java.util.Date;
+
+import org.apache.james.blob.api.BlobId;
+import org.apache.james.mailbox.model.Content;
+import org.apache.james.mailbox.model.MessageId;
+
+import com.google.common.base.Preconditions;
+
+public class MessageRepresentation {
+    public static MessageRepresentation.Builder builder() {
+        return new MessageRepresentation.Builder();
+    }
+
+    public static class Builder {
+        private MessageId messageId;
+        private Date internalDate;
+        private Long size;
+        private Content headerContent;
+        private BlobId bodyBlobId;
+
+        public MessageRepresentation.Builder messageId(MessageId messageId) {
+            this.messageId = messageId;
+            return this;
+        }
+
+        public MessageRepresentation.Builder internalDate(Date internalDate) {
+            this.internalDate = internalDate;
+            return this;
+        }
+
+        public MessageRepresentation.Builder size(long size) {
+            Preconditions.checkArgument(size >= 0, "size can not be negative");
+            this.size = size;
+            return this;
+        }
+
+        public MessageRepresentation.Builder headerContent(Content headerContent) {
+            this.headerContent = headerContent;
+            return this;
+        }
+
+        public MessageRepresentation.Builder bodyBlobId(BlobId bodyBlobId) {
+            this.bodyBlobId = bodyBlobId;
+            return this;
+        }
+
+        public MessageRepresentation build() {
+            Preconditions.checkNotNull(messageId, "messageId is required");
+            Preconditions.checkNotNull(internalDate, "internalDate is required");
```

Comment:
Q:
I see the `internal_date` in the database accepts NULL value
is it ok when checkNotNull here?

Reply:
I did not know that. Q: In which chance `internal_date` could be null?

Reply:
> I did not know that. Q: In which chance `internal_date` could be null?

yes, It can be null

Reply:
FYI, `SimpleMailboxMessage:build()` (which Postgres is using to build `SimpleMailboxMessage`) requires `internalDate`.

IMO `internalDate` or even `saveDate` should not be null in the `message` table.

For `saveDate` in Cassandra, it was possiblly null because of retro-compatibility (no saveDate column before), while it is not the case for Postgres.